src/main/java/com/mattermost/integration/figma/notification/service/FileNotificationService.java

```diff
@@ -0,0 +1,58 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class FileNotificationService {
+    private static final String BASE_WEBHOOK_URL = "https://api.figma.com/v2/webhooks";
+    private static final String PASSCODE = "Mattermost";
+    private static final String FILE_COMMENT_EVENT_TYPE = "FILE_COMMENT";
+    //Production Mattermost link
+    private static final String REDIRECT_URL = "http://localhost:8066/plugins/com.mattermost.apps/apps/spring-boot-example%s?secret=%s";
+    private static final String FILE_COMMENT_URL = "/webhook/comment";
+
+    private final RestTemplate restTemplate;
+
+    public FileNotificationService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public String subscribeToFileNotification(String teamId, String webhookSecret) {
+        if (teamId != null && !teamId.isEmpty() && !teamId.isBlank()) {
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            headers.set("Authorization", String.format("Bearer %s", FigmaTokenProvider.token.getAccessToken()));
+            FileCommentNotificationRequest fileCommentNotificationRequest = new FileCommentNotificationRequest();
+            fileCommentNotificationRequest.setEventType(FILE_COMMENT_EVENT_TYPE);
+            fileCommentNotificationRequest.setTeamId(teamId);
+            fileCommentNotificationRequest.setPasscode(PASSCODE);
+            //For production Mattermost link
+            //fileCommentNotificationRequest.setEndpoint(String.format(REDIRECT_URL, FILE_COMMENT_URL, webhookSecret));
+            fileCommentNotificationRequest.setEndpoint(NgrokLinkProvider.REDIRECT_URL.concat(FILE_COMMENT_URL));
+            System.out.println(fileCommentNotificationRequest);
+            HttpEntity<FileCommentNotificationRequest> request = new HttpEntity<>(fileCommentNotificationRequest, headers);
+            restTemplate.postForEntity(BASE_WEBHOOK_URL, request, String.class);
+            return "{\"text\" : \"Success\"}";
+        }
+        return "{\"text\" : \"There is no such team id\"}";
+    }
+
+    public void deleteWebhook(String webhookId) {
+        MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+        headers.add("Authorization", String.format("Bearer %s", FigmaTokenProvider.token.getAccessToken()));
+        HttpEntity<Object> request = new HttpEntity<>(headers);
+        String url = BASE_WEBHOOK_URL.concat("/").concat(webhookId);
+        restTemplate.exchange(url, HttpMethod.DELETE, request, String.class);
+        System.out.println("Successfully deleted webhook with id: " + webhookId);
```

Comment:
hm, is it okay if we add a logger in this PR  insead of using sout ? Or we can do it in a another PR